data/libs/SpaceStation.lua

```diff
@@ -36,16 +40,26 @@ function SpaceStation:Constructor()
 end
 
 local equipmentStock = {}
+local FlightLog = require 'FlightLog'
 
 local function updateEquipmentStock (station)
 	assert(station and station:exists())
 	if equipmentStock[station] then return end
 	equipmentStock[station] = {}
 
+	local timediff = 0
+	for systemp,arrtime,deptime in FlightLog.GetSystemPaths() do
+		local time_interval = Game.time - (deptime or Game.time)
+		if time_interval < 60 * 30 * 24 * 60 * 60 and Game.system.name == systemp:GetStarSystem().name then	-- After 5 years of absence systems consider player new customer
+			timediff = math.max(time_interval, timediff)
+		end
+	end
+	local recentSystem = timediff < 6 * 30 * 24 * 60 * 60
```

Comment:
I like the idea of tracking system stock over time to make it seem like there is a depletion effect happening from the player repeatedly trading from it, but I'd personally want to use something like a random walk to calculate the current stock amount when the player docks over some interval of weeks. I'd have to triple check how station restocking works right now, but if I am correct this particular code might be better off replaced with code that tracks the station's stock history instead of statically setting the stock to a certain value.

Reply:
I am happy you liked the idea. My main guide line, when i started all of this, was to use no new code but to explore the possibilities that are present with current code.
You can alter it, but should consider my experience so far. That is that small variations will not be detectable by players because of large variations in stock values on export modes. I rejected my thought for periodical solutions like sin(time) because of their continued feature. I can not follow the idea of random walk but i believe it will introduce graduate depletion that will act as some continues function. Orbit seasons will not work i think for the same reason and in some cases they might not work at all, as for examble when orbiting period are years or decades. Those solutions have the problem that use game time while player uses, through time acceleration, his personal time. I think it is ambiguous the claim that a player will use game time scheduled series of travels, beyond his next one.

This block of code do not let me rest in my games and proved its worth to me.

Having said that, it has nothing to do with exponential and the main scope of this PR. Modify it at will, i would prefer though to test it in action before :eyes: